app/controllers/locations_controller.rb

```diff
@@ -1,25 +1,29 @@
 class LocationsController < ApplicationController
   def create
-    @location = build_location
+    @profile = current_user
+
+    if save_location?
+      enqueue_geocode_job
 
-    if update_location?
       redirect_to profile_url
     else
-      @profile = current_user
-
       render "profiles/show"
```

Comment:
Oh great! I see in the template you're using `@profile.location` already; I thought it was using the `@location` instance. Do you feel like this is pretty clean now?